backend/app/controllers/spree/admin/shipping_methods_controller.rb

```diff
@@ -5,6 +5,26 @@ class ShippingMethodsController < ResourceController
       before_action :set_shipping_category, only: [:create, :update]
       before_action :set_zones, only: [:create, :update]
 
+      def update
+        invoke_callbacks(:update, :before)
+        if update_calculator_attributes && @object.update(permitted_shipping_method_params)
```

Comment:
When I changed currency field to selector, `calculator_attributes` params come with each request.
When updating only calculator attributes it works fine, but  when updating calculator it throws error:

> Failure/Error: if @object.update(permitted_resource_params)
>ActiveModel::UnknownAttributeError:
  unknown attribute 'preferred_currency' for Spree::Calculator.
./app/controllers/spree/admin/resource_controller.rb:27:in `update'

However when I blacklist `calculator_attributes` I can't change currency.

My solution is adding `update_calculator_attributes` method so when `Spree::Calculator` is not changed, it's atrributed is also updated.